code/modules/vending/vendor/interaction.dm

```diff
@@ -0,0 +1,234 @@
+		to_chat(user, span_warning("[src] does not accept [inserted_item]!"))
+		return FALSE
+
+	to_chat(user, span_notice("You insert [inserted_item] into [src]'s input compartment."))
+	for(var/datum/data/vending_product/product_datum in product_records + coin_records + hidden_records)
+		if(inserted_item.type == product_datum.product_path)
+			if(product_datum.amount == product_datum.max_amount)
+				to_chat(user, span_warning("no space for any more [product_datum.category || "Products"]!"))
+				return FALSE
+
+			if(!user.transferItemToLoc(inserted_item, src))
+				to_chat(user, span_warning("[inserted_item] is stuck in your hand!"))
+				return FALSE
+
+			product_datum.amount++
+			LAZYADD(product_datum.returned_products, inserted_item)
+			break
+
+/obj/machinery/vending/item_interaction(mob/living/user, obj/item/attack_item, list/modifiers)
+	. = NONE
+	if(panel_open && is_wire_tool(attack_item))
+		wires.interact(user)
+		return ITEM_INTERACT_SUCCESS
+
+	if(refill_canister && istype(attack_item, refill_canister))
+		if (!panel_open)
+			to_chat(user, span_warning("You should probably unscrew the service panel first!"))
+		else if (!is_operational)
+			to_chat(user, span_notice("[src] does not respond."))
+		else
+			//if the panel is open we attempt to refill the machine
+			var/obj/item/vending_refill/canister = attack_item
+			if(canister.get_part_rating() == 0)
+				to_chat(user, span_warning("[canister] is empty!"))
+			else
+				// instantiate canister if needed
+				post_restock(user, restock(canister))
+			return ITEM_INTERACT_SUCCESS
+
+	if(compartmentLoadAccessCheck(user) && !user.combat_mode)
+		if(istype(attack_item, /obj/item/storage/bag)) //trays USUALLY
+			var/obj/item/storage/storage_item = attack_item
+			var/loaded = 0
+			var/denied_items = 0
+			for(var/obj/item/the_item in storage_item.contents)
+				if(contents.len >= MAX_VENDING_INPUT_AMOUNT) // no more than 30 item can fit inside, legacy from snack vending although not sure why it exists
+					to_chat(user, span_warning("[src]'s compartment is full."))
+					break
+				if(loadingAttempt(the_item, user))
+					loaded++
+				else
+					denied_items++
+			if(denied_items)
+				to_chat(user, span_warning("[src] refuses some items!"))
+			if(loaded)
+				to_chat(user, span_notice("You insert [loaded] dishes into [src]'s compartment."))
+			return ITEM_INTERACT_SUCCESS
+		else
+			return loadingAttempt(attack_item, user) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_FAILURE
+
+	if(tiltable && !tilted && attack_item.force)
+		if(isclosedturf(get_turf(user))) //If the attacker is inside of a wall, immediately fall in the other direction, with no chance for goodies.
+			tilt(get_turf(get_step(src, REVERSE_DIR(get_dir(src, user)))))
+			return ITEM_INTERACT_SUCCESS
+
+		switch(rand(1, 100))
+			if(1 to 5)
+				freebie(3)
+			if(6 to 15)
+				freebie(2)
+			if(16 to 25)
+				freebie(1)
+			if(26 to 75)
+				return
+			if(76 to 100)
+				tilt(user)
+		return ITEM_INTERACT_SUCCESS
```

Comment:
This should definitely still be in attackby - dispensing free stuff is an effect of an attack, not an item interaction

Reply:
So I move out only the freebe code.

Reply:
This code should only run after being hit yeah